src/rust/engine/fs/src/store.rs

```diff
@@ -318,10 +334,20 @@ impl Store {
                 })
             })
             .collect::<Vec<_>>(),
-        )
+        ).and_then(future::join_all)
+          .map(|uploaded_digests| (uploaded_digests, ingested_digests))
+      })
+      .map(|(uploaded_digests, ingested_digests)| {
+        let ingested_file_sizes = ingested_digests.iter().map(|(digest, _)| digest.1);
+        let uploaded_file_sizes = uploaded_digests.iter().map(|digest| digest.1);
+
+        UploadSummary {
+          ingested_file_count: ingested_file_sizes.clone().count(),
```

Comment:
Neither of these iterators filter, so it would be constant time to just do `ingested_digests.len()` to get the count.